contrib/opencensus-ext-flask/opencensus/ext/flask/flask_middleware.py

```diff
@@ -179,6 +176,9 @@ def _after_request(self, response):
 
         try:
             tracer = execution_context.get_opencensus_tracer()
+            tracer.add_attribute_to_current_span(
+                HTTP_ROUTE, flask.request.url_rule.rule
```

Comment:
Note that this will throw if url_rule is None, which it will for any 404 route. I have raised https://github.com/census-instrumentation/opencensus-python/issues/777 for this